bitcoin/base58.c

```diff
@@ -12,26 +12,24 @@
 #include <ccan/build_assert/build_assert.h>
 #include <ccan/tal/str/str.h>
 #include <common/utils.h>
-#include <libbase58.h>
 #include <string.h>
-
-static bool my_sha256(void *digest, const void *data, size_t datasz)
-{
-	sha256(digest, data, datasz);
-	return true;
-}
+#include <wally_core.h>
 
 static char *to_base58(const tal_t *ctx, u8 version,
 		       const struct ripemd160 *rmd)
 {
-	char out[BASE58_ADDR_MAX_LEN + 1];
-	size_t outlen = sizeof(out);
+	char *out;
+	size_t total_length = sizeof(*rmd) + 1;
+	u8 buf[total_length];
+	buf[0] = version;
+	memcpy(buf + 1, rmd, sizeof(*rmd));
 
-	b58_sha256_impl = my_sha256;
-	if (!b58check_enc(out, &outlen, version, rmd, sizeof(*rmd))) {
+	if (wally_base58_from_bytes((const unsigned char *) buf, total_length, BASE58_FLAG_CHECKSUM, &out) != WALLY_OK) {
 		return NULL;
 	}else{
-		return tal_strdup(ctx, out);
+		char *res;
+		res = tal_strdup(ctx, out);
+		return wally_free_string(out) == WALLY_OK ? res : NULL;
```

Comment:
This may be considered a leak by the leak detector since we return `NULL` without freeing `out`.

Reply:
Is this ok now?